### PR TITLE
Implementation of Floating IP Address resource inc Tests and Website docs

### PR DESCRIPTION
**Pull request by trentrosenbaum**
Implements the Floating IP Address resource.  The resource will allow a floating IP address to be provisioned either attached or detached.  The PR provides the resource, tests and website docs.

The acceptance tests cover the creation of a detached resource and then an attached resource followed by a floating IP that is switched between servers/network interfaces.  The final test covers the import of an IP address.

### CODE REVIEW

File: upcloud/resource_upcloud_floating_ip_address.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import (
+	"context"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceUpCloudFloatingIpAddress() *schema.Resource {
```
Comment (yoink00) [inline upcloud/resource_upcloud_floating_ip_address.go:12]: <nitpick>My linting complains that "Ip" should be "IP"...</nitpick>

File: upcloud/resource_upcloud_floating_ip_address.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+func resourceUpCloudFloatingIpAddressRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*service.Service)
+
+	var diags diag.Diagnostics
+
+	getIPAddressDetailsRequest := &request.GetIPAddressDetailsRequest{
+		Address: d.Id(),
+	}
+
+	ipAdress, err := client.GetIPAddressDetails(getIPAddressDetailsRequest)
```
Comment (yoink00) [inline upcloud/resource_upcloud_floating_ip_address.go:97]: ipA**d**ress should be ipA**dd**ress

File: upcloud/resource_upcloud_floating_ip_address.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func resourceUpCloudFloatingIpAddressUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*service.Service)
+
+	modifyIPAddressRequest := &request.ModifyIPAddressRequest{
+		IPAddress: d.Id(),
+	}
+
+	if d.HasChange("access") {
```
Comment (yoink00) [inline upcloud/resource_upcloud_floating_ip_address.go:133]: The should be "mac_address" not "access".

Comment (trentrosenbaum) [reply upcloud/resource_upcloud_floating_ip_address.go]: This has been updated.

File: upcloud/resource_upcloud_floating_ip_address.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+func resourceUpCloudFloatingIPAddressRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*service.Service)
+
+	var diags diag.Diagnostics
+
+	getIPAddressDetailsRequest := &request.GetIPAddressDetailsRequest{
+		Address: d.Id(),
+	}
+
+	ipAdress, err := client.GetIPAddressDetails(getIPAddressDetailsRequest)
```
Comment (yoink00) [inline upcloud/resource_upcloud_floating_ip_address.go:98]: `ipAdress` is spelt incorrectly.

**Review by yoink00 — APPROVED** [review]
I've left a comment about a spelling mistake. I've approved it so you don't have to wait for me once that's fixed 😸

**Pull request merged**